### PR TITLE
DCS-1756  Update deps , java and gradle

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -17,12 +17,12 @@ dependencies {
...
-  implementation("org.springdoc:springdoc-openapi-webmvc-core:1.6.8")
-  implementation("org.springdoc:springdoc-openapi-ui:1.6.8")
-  implementation("org.springdoc:springdoc-openapi-kotlin:1.6.8")
+  implementation("org.springdoc:springdoc-openapi-webmvc-core:1.6.9")
+  implementation("org.springdoc:springdoc-openapi-ui:1.6.9")
+  implementation("org.springdoc:springdoc-openapi-kotlin:1.6.9")
 
   runtimeOnly("com.h2database:h2:2.1.214")
-  runtimeOnly("org.flywaydb:flyway-core:8.5.13")
+  runtimeOnly("org.flywaydb:flyway-core")
```
Comment (andrewrlee) [inline build.gradle.kts:25]: now relying on provided version

File: build.gradle.kts [hunk 2]
```diff
@@ -40,13 +40,13 @@ allOpen {
...
 java {
-  toolchain.languageVersion.set(JavaLanguageVersion.of(17))
+  toolchain.languageVersion.set(JavaLanguageVersion.of(18))
 }
 
 tasks {
   withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
     kotlinOptions {
-      jvmTarget = "17"
+      jvmTarget = "18"
```
Comment (andrewrlee) [inline build.gradle.kts:49]: bumping java

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -56,6 +55,9 @@ spring:
       resourceserver:
         jwt:
           jwk-set-uri: ${oauth.endpoint.url}/.well-known/jwks.json
+  sql:
+    init:
+      continue-on-error: true
```
Comment (andrewrlee) [inline src/main/resources/application.yml:60]: moving away from deprecated property

File: src/test/java/uk/gov/justice/hmpps/probationteams/repository/LocalDeliveryUnitRepositoryTest.kt [hunk 4]
```diff
@@ -17,8 +17,8 @@ import uk.gov.justice.hmpps.probationteams.utils.uniqueLduCode
 import java.util.UUID
 
 @ActiveProfiles("test")
-@ExtendWith(SpringExtension::class)
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE)
+@EnableAutoConfiguration(exclude = [SpringDocKotlinConfiguration::class])
```
Comment (andrewrlee) [inline src/test/java/uk/gov/justice/hmpps/probationteams/repository/LocalDeliveryUnitRepositoryTest.kt:21]: this was causing an issue in latest springdoc as required beans were not included when web env is NONE

File: build.gradle.kts (hunk 2 above)
Comment (neilmendum) [reply build.gradle.kts:49]: Think you'll also want to bump the `Dockerfile` JDK version?

**Review by GurnankCheema — APPROVED** [review]
LGTM

**Review by neilmendum — APPROVED** [review]
LGTM

File: build.gradle.kts [hunk 5]
```diff
@@ -1,5 +1,5 @@
 plugins {
-  id("uk.gov.justice.hmpps.gradle-spring-boot") version "4.3.2"
+  id("uk.gov.justice.hmpps.gradle-spring-boot") version "4.4.1-beta"
```
Comment (thirlz95) [inline build.gradle.kts:2]: You may want to wait till this is outta beta, otherwise you'll be changing it again very shortly

**Pull request merged**